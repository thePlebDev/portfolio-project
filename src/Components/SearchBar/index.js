import React,{useState,useEffect} from 'react'

import styled from 'styled-components';
import PropTypes from 'prop-types';

import useSearchInput from '../../Hooks/useSearchInput'
import SearchInput from '../SearchInput';
import SearchResults from '../SearchResults'

const Container = styled.div`
  position:absolute;
  width:100%;
  z-index:999999999999;
`




const SearchBar = ({show,setShow})=>{
  const [searchResults,setSearchResults] = useState()

  useEffect(()=>{
    let cachedHits = sessionStorage.getItem('posts')
    setSearchResults(JSON.parse(cachedHits))
  },[])

  const {state,handleChange} = useSearchInput(searchResults,setSearchResults)
  return(
    <Container>
      {
        show
          ?
            <div data-testid="showTrue">
                <SearchInput state={state.search} handleChange={handleChange} setSearchResults={setSearchResults} searchResults={searchResults} setShow={setShow}/>
                <SearchResults searchResults={searchResults}/>
            </div>
          :
            <div data-testid="showFalse"></div>
    }
    </Container>
  )

}

SearchBar.propTypes ={
  show:PropTypes.bool,
  setShow:PropTypes.func
}
SearchBar.defaultProps ={
  show:false,
  setShow:()=>{console.log('ENTER A PROPER SET SHOW FUNCTION')}
}

export default SearchBar
