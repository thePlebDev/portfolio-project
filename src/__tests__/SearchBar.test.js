import React from 'react';

import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import SearchBar from '../Components/SearchBar';

const findByAttr =(component,dataTestId)=>{
  const data = component.find(`[data-testid='${dataTestId}']`)
  return data
}


describe('testing the SearchBar component',()=>{
  it('should test the snapshot',()=>{
    const tree = renderer.create(<SearchBar/>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render the important stuff with the default props',()=>{
    const wrapper = shallow(<SearchBar/>)
    //showFalse
    expect(findByAttr(wrapper,'showFalse').length).toBe(1)
  })
  it('should render the important stuff with the custom props',()=>{
    const show = true;
    const wrapper = shallow(<SearchBar show={show}/>)
    expect(findByAttr(wrapper,'showTrue').length).toBe(1)
  })
})
