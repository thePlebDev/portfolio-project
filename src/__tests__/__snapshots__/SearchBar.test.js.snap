// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the SearchBar component should test the snapshot 1`] = `
<div
  className="sc-pNWxx jSDNas"
>
  <div
    data-testid="showFalse"
  />
</div>
`;
